### PR TITLE
D2L0051: ConfigViewerAnalyzer to ban usage of certain config variables

### PR DESCRIPTION
**Pull request by omsmith**
- [x] Finish cleanup in LMS

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/Diagnostics.cs [hunk 1]
```diff
@@ -434,5 +434,15 @@ public static class Diagnostics {
 			isEnabledByDefault: true,
 			description: "The parameter {0} has a default value of {1} here, but {2} in its original definition in {3}. This causes inconsistent behaviour. Please use the same defualt value everywhere."
 		);
+
+		public static readonly DiagnosticDescriptor BannedConfig = new DiagnosticDescriptor(
+			id: "D2LXXXX",
+			title: "Fetching the config variable \"{0}\" has been banned. {1}",
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Diagnostics.cs]: I wanted a "nicer" word than banned, but couldn't think of one, outside of deprecated? I'll just do deprecated

**Comment by j3parker** [comment]
Do you plan to program in a whitelist to this?

**Comment by j3parker** [comment]
Nit: column lengths greater than 80 cols get kind of annoying to review on GitHub.

**Comment by omsmith** [comment]
I was thinking of just ignoring the diagnostic in the few places that need it, which I think will be down to ~2

**Comment by j3parker** [comment]
Cool. That's simple.

**Pull request merged**